src/fusiontensor/base_interface.jl

```diff
@@ -0,0 +1,165 @@
+# This files defines Base functions for FusionTensor
+
+using BlockSparseArrays: @view!
+
+function Base.:*(x::Number, ft::FusionTensor)
+  return FusionTensor(
+    x * data_matrix(ft), codomain_axes(ft), domain_axes(ft), trees_block_mapping(ft)
+  )
```

Comment:
This and other functions can be simplified with the follow "field setting" code pattern:
```julia
using Accessors: @set
set_data_matrix(ft::FusionTensor, data_matrix) = @set ft.data_matrix = data_matrix
Base.:*(x::Number, ft::FusionTensor) = set_data_matrix(ft, x * data_matrix(ft))
```

Reply:
In general I like to define field getter/setter functions like that for the fields of types I define since they come in handy and can simplify a lot of code, and make for better generic code.